### PR TITLE
Use ConnectionManager

### PR DESCRIPTION
**Pull request by justmobilize**
Update `adafruit_requests` to use [Adafruit_CircuitPython_ConnectionManager](https://github.com/adafruit/Adafruit_CircuitPython_ConnectionManager/) for socket handling. As well as clean up examples

### CODE REVIEW

**Comment by justmobilize** [comment]
Updated the RTD example section to have all 4 boards listed:
image

**Comment by justmobilize** [comment]
@tannewt asking here, just so we have record of the conversation.

The code that is currently [here](https://github.com/adafruit/Adafruit_CircuitPython_Requests/blob/c533f184d7e1af70a8aac281055f38f555c93641/adafruit_requests.py#L619C1-L680C57) which is re-worked from the old `set_socket` to set a global instance - Is this something we want to keep?

Since `set_socket` is gone, everyone will still need to move to passing in a `socket_pool` and `ssl_context` and thus aren't making anything easier by leaving this here.

**Comment by justmobilize** [comment]
@tannewt I'm also moving this to `ready for review`. I would like to add some more tests before it merges one we know what's what

**Comment by tannewt** [comment]
> @tannewt asking here, just so we have record of the conversation.
> 
> The code that is currently [here](https://github.com/adafruit/Adafruit_CircuitPython_Requests/blob/c533f184d7e1af70a8aac281055f38f555c93641/adafruit_requests.py#L619C1-L680C57) which is re-worked from the old `set_socket` to set a global instance - Is this something we want to keep?
> 
> Since `set_socket` is gone, everyone will still need to move to passing in a `socket_pool` and `ssl_context` and thus aren't making anything easier by leaving this here.

Ya, let's just remove it in favor of Session().

File: examples/wifi/expanded/requests_wifi_api_github.py [hunk 1]
```diff
@@ -22,11 +20,12 @@
...
 sleep_time = 900
 
-try:
-    from secrets import secrets
-except ImportError:
-    print("Secrets File Import Error")
-    raise
+# Get WiFi details, ensure these are setup in settings.toml
+ssid = os.getenv("CIRCUITPY_WIFI_SSID")
+appw = os.getenv("CIRCUITPY_WIFI_PASSWORD")
```
Comment (tannewt) [inline examples/wifi/expanded/requests_wifi_api_github.py:25]: What is `appw` for?

**Comment by justmobilize** [comment]
> > @tannewt asking here, just so we have record of the conversation.
> > The code that is currently [here](https://github.com/adafruit/Adafruit_CircuitPython_Requests/blob/c533f184d7e1af70a8aac281055f38f555c93641/adafruit_requests.py#L619C1-L680C57) which is re-worked from the old `set_socket` to set a global instance - Is this something we want to keep?
> > Since `set_socket` is gone, everyone will still need to move to passing in a `socket_pool` and `ssl_context` and thus aren't making anything easier by leaving this here.
> 
> Ya, let's just remove it in favor of Session().

@tannewt after removing and adding tests, any other changes?

File: examples/wifi/expanded/requests_wifi_api_github.py (hunk 1 above)
Comment (justmobilize) [reply examples/wifi/expanded/requests_wifi_api_github.py:25]: I just standardized in what there was the most of. Assumed it was an internal preference. Can change them all to `password` or anything else that makes sense.

Comment (justmobilize) [reply examples/wifi/expanded/requests_wifi_api_github.py:25]: Ahhh- `AP PassWord`

Comment (tannewt) [reply examples/wifi/expanded/requests_wifi_api_github.py:25]: I'd prefer `password`. This is the only other change from me.

**Comment by justmobilize** [comment]
@tannewt this is ready for review again. Coverage is at 85%, I tackled what was easy and added some fixtures to remove boilerplate. Will add more tests in a future pass.

image

**Review by tannewt — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**